staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -187,18 +188,11 @@ func (o objectTrackerReact) Create(action CreateActionImpl) (runtime.Object, err
 func (o objectTrackerReact) Update(action UpdateActionImpl) (runtime.Object, error) {
 	ns := action.GetNamespace()
 	gvr := action.GetResource()
-	objMeta, err := meta.Accessor(action.GetObject())
-	if err != nil {
-		return nil, err
-	}
-
-	err = o.tracker.Update(gvr, action.GetObject(), ns, action.UpdateOptions)
+	err := o.tracker.Update(gvr, action.GetObject(), ns, action.UpdateOptions)
 	if err != nil {
 		return nil, err
 	}
-
-	obj, err := o.tracker.Get(gvr, ns, objMeta.GetName(), metav1.GetOptions{})
-	return obj, err
+	return action.GetObject(), err
```

Comment:
Sadly, `tracker.Update()` does not return the updated object, and we cannot get the object from the tracker again because it may already be deleted at `Update`. So we just return the passed in object, which is not a deep copy, and may lack of namespace that is populated in `Update`.